beekeeper-core/src/test/java/com/expediagroup/beekeeper/core/repository/HousekeepingMetadataRepositoryTest.java

```diff
@@ -252,12 +252,85 @@ public void findRecordForCleanupByDatabaseAndTableMixedHousekeepingStatus() {
     compare(result.get(), housekeepingTable1);
   }
 
-  private HousekeepingMetadata createEntityHousekeepingTable() {
+  @Test
+  public void findRecordsForGivenDatabaseAndTable() {
+    HousekeepingMetadata housekeepingTable1 = createPartitionedEntityHousekeepingTable();
+    housekeepingMetadataRepository.save(housekeepingTable1);
+
+    HousekeepingMetadata housekeepingTable2 = createUnpartitionedEntityHousekeepingTable();
+    housekeepingMetadataRepository.save(housekeepingTable2);
+
+    Page<HousekeepingMetadata> result = housekeepingMetadataRepository
+        .findRecordsForGivenDatabaseAndTable(DATABASE_NAME, TABLE_NAME, PageRequest.of(0, 500));
+
+    assertTrue(result.getContent().size() == 2);
+    compare(result.getContent().get(0), housekeepingTable1);
+    compare(result.getContent().get(1), housekeepingTable2);
+  }
+
+  @Test
+  public void findRecordsForGivenDatabaseAndTableEmpty() {
+    HousekeepingMetadata housekeepingTable1 = createEntityHouseKeepingTable(DATABASE_NAME + "1", TABLE_NAME + "1",
+        PARTITION_NAME);
+    housekeepingMetadataRepository.save(housekeepingTable1);
+
+    Page<HousekeepingMetadata> result = housekeepingMetadataRepository
+        .findRecordsForGivenDatabaseAndTable(DATABASE_NAME, TABLE_NAME, PageRequest.of(0, 500));
+
+    assertTrue(result.isEmpty());
+  }
+
+  @Test
+  public void findRecordsForGivenDatabaseAndTableMultipleDatabaseEntries() {
+    HousekeepingMetadata housekeepingTable1 = createPartitionedEntityHousekeepingTable();
+    housekeepingMetadataRepository.save(housekeepingTable1);
+    HousekeepingMetadata housekeepingTable2 = createEntityHouseKeepingTable(DATABASE_NAME + "1", TABLE_NAME + "1",
+        PARTITION_NAME);
+    housekeepingMetadataRepository.save(housekeepingTable2);
+
+    Page<HousekeepingMetadata> result = housekeepingMetadataRepository
+        .findRecordsForGivenDatabaseAndTable(DATABASE_NAME, TABLE_NAME, PageRequest.of(0, 500));
+
+    assertTrue(result.getContent().size() == 1);
```

Comment:
Rather use `assertEquals` or `assertThat....` as one gets a better failure message if this isn't true.